Infinispan client with disabled devservices  failing to start in dev mode

### Describe the bug

When running the devmode with disabled devservices (`quarkus.devservices.enabled=false`) and the app contains `quarkus-infinispan-client`.

The error throws ` Null contextual instance was produced by a normal scoped synthetic bean: SYNTHETIC bean [class=org.infinispan.client.hotrod.RemoteCacheManager, id=EsszMmOX-DojxN4A7ia-MjyfeSA]`

### Expected behavior

Start the Quarkus in dev mode without problem or with error which tell the user more.

### Actual behavior

```
 Failed to start application: java.lang.RuntimeException: Failed to start quarkus
        at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
        at io.quarkus.runtime.Application.start(Application.java:116)
        at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:127)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:79)                                                                                                                                                                                                                                                                                                                    
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:50)                                                                                                                                                                                                                                                                                                                    
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:143)                                                                                                                                                                                                                                                                                                                   
        at io.quarkus.runner.GeneratedMain.main(Unknown Source)                                                                                                                                                                                                                                                                                                               
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)                                                                                                                                                                                                                                                              
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)                                                                                                                                                                                                                                                                                                         
        at io.quarkus.runner.bootstrap.StartupActionImpl$1.run(StartupActionImpl.java:154)                                                                                                                                                                                                                                                                                    
        at java.base/java.lang.Thread.run(Thread.java:1583)                                                                                                                                                                                                                                                                                                                   
Caused by: jakarta.enterprise.inject.CreationException: Null contextual instance was produced by a normal scoped synthetic bean: SYNTHETIC bean [class=org.infinispan.client.hotrod.RemoteCacheManager, id=EsszMmOX-DojxN4A7ia-MjyfeSA]                                                                                                                                       
        at org.infinispan.client.hotrod.RemoteCacheManager_EsszMmOX-DojxN4A7ia-MjyfeSA_Synthetic_Bean.create(Unknown Source)                                                                                                                                                                                                                                                  
        at org.infinispan.client.hotrod.RemoteCacheManager_EsszMmOX-DojxN4A7ia-MjyfeSA_Synthetic_Bean.create(Unknown Source)                                                                                                                                                                                                                                                  
        at io.quarkus.arc.impl.AbstractSharedContext.createInstanceHandle(AbstractSharedContext.java:119)                                                                                                                                                                                                                                                                     
        at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:38)                                                                                                                                                                                                                                                                                     
        at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:35)                                                                                                                                                                                                                                                                                     
        at io.quarkus.arc.generator.Default_jakarta_enterprise_context_ApplicationScoped_ContextInstances.ch7(Unknown Source)                                                                                                                                                                                                                                                 
        at io.quarkus.arc.generator.Default_jakarta_enterprise_context_ApplicationScoped_ContextInstances.computeIfAbsent(Unknown Source)                                                                                                                                                                                                                                     
        at io.quarkus.arc.impl.AbstractSharedContext.get(AbstractSharedContext.java:35)                                                                                                                                                                                                                                                                                       
        at io.quarkus.arc.impl.ClientProxies.getApplicationScopedDelegate(ClientProxies.java:23)                                                                                                                                                                                                                                                                              
        at org.infinispan.client.hotrod.RemoteCacheManager_EsszMmOX-DojxN4A7ia-MjyfeSA_Synthetic_ClientProxy.arc$delegate(Unknown Source)                                                                                                                                                                                                                                     
        at org.infinispan.client.hotrod.RemoteCacheManager_EsszMmOX-DojxN4A7ia-MjyfeSA_Synthetic_ClientProxy.getConfiguration(Unknown Source)                                                                                                                                                                                                                                 
        at io.quarkus.infinispan.client.runtime.InfinispanRecorder.initializeClient(InfinispanRecorder.java:70)                                                                                                                                                                                                                                                               
        at io.quarkus.runner.recorded.InfinispanClientProcessor$infinispanClients405495887.deploy_0(Unknown Source)                                                                                                                                                                                                                                                           
        at io.quarkus.runner.recorded.InfinispanClientProcessor$infinispanClients405495887.deploy(Unknown Source)                                                                                                                                                                                                                                                             
        ... 11 more
```

### How to Reproduce?

1. `git clone -b infinispan-disable-devservices https://github.com/jedla97/quarkus-reproducers.git`
2. `cd quarkus-reproducers`
3. ` quarkus dev -Dquarkus.devservices.enabled=false`
4. devmode fail to start and the error is present

With Quarkus 3.31.2 it works without problem (`quarkus dev -Dquarkus.devservices.enabled=false -Dquarkus.platform.version=3.31.2`)

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

25.0.2-tem, 21.0.9-tem

### Quarkus version or git rev

main

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

Note: We have test which create app with multiple extensions and start it in devmode without devservices just to check it's possible to create, build and run the app.

--- a/extensions/infinispan-client/runtime/src/main/java/io/quarkus/infinispan/client/runtime/InfinispanRecorder.java
+++ b/extensions/infinispan-client/runtime/src/main/java/io/quarkus/infinispan/client/runtime/InfinispanRecorder.java
@@ -1,9 +1,7 @@
 package io.quarkus.infinispan.client.runtime;
 
-import java.util.HashSet;
 import java.util.Map;
 import java.util.Properties;
-import java.util.Set;
 import java.util.function.Function;
 import java.util.function.Supplier;
 
@@ -31,10 +29,7 @@ public class InfinispanRecorder {
         };
     }
 
-    private final Set<String> clientNames = new HashSet<>();
-
     public Supplier<RemoteCacheManager> infinispanRemoteCacheManagerSupplier(String clientName) {
-        clientNames.add(clientName);
         return new InfinispanClientSupplier<>(new Function<>() {
             @Override
             public RemoteCacheManager apply(InfinispanClientProducer infinispanClientProducer) {
@@ -44,7 +39,6 @@ public class InfinispanRecorder {
     }
 
     public Supplier<CounterManager> infinispanCounterManagerSupplier(String clientName) {
-        clientNames.add(clientName);
         return new InfinispanClientSupplier<>(new Function<>() {
             @Override
             public CounterManager apply(InfinispanClientProducer infinispanClientProducer) {
@@ -54,7 +48,6 @@ public class InfinispanRecorder {
     }
 
     public <K, V> Supplier<RemoteCache<K, V>> infinispanRemoteCacheSupplier(String clientName, String cacheName) {
-        clientNames.add(clientName);
         return new InfinispanClientSupplier<>(new Function<>() {
             @Override
             public RemoteCache<K, V> apply(InfinispanClientProducer infinispanClientProducer) {

--- a/extensions/infinispan-client/runtime/src/main/java/io/quarkus/infinispan/client/runtime/InfinispanRecorder.java
+++ b/extensions/infinispan-client/runtime/src/main/java/io/quarkus/infinispan/client/runtime/InfinispanRecorder.java
@@ -14,6 +14,7 @@ import org.infinispan.client.hotrod.RemoteCacheManager;
 import org.infinispan.counter.api.CounterManager;
 
 import io.quarkus.arc.Arc;
+import io.quarkus.arc.runtime.BeanContainer;
 import io.quarkus.arc.runtime.BeanContainerListener;
 import io.quarkus.runtime.RuntimeValue;
 import io.quarkus.runtime.annotations.Recorder;
@@ -23,9 +24,12 @@ import io.quarkus.runtime.annotations.RelaxedValidation;
 public class InfinispanRecorder {
 
     public BeanContainerListener configureInfinispan(@RelaxedValidation Map<String, Properties> properties) {
-        return container -> {
-            InfinispanClientProducer instance = container.beanInstance(InfinispanClientProducer.class);
-            instance.setProperties(properties);
+        return new BeanContainerListener() {
+            @Override
+            public void created(BeanContainer container) {
+                InfinispanClientProducer instance = container.beanInstance(InfinispanClientProducer.class);
+                instance.setProperties(properties);
+            }
         };
     }
 

--- a/extensions/infinispan-client/runtime/src/main/java/io/quarkus/infinispan/client/runtime/InfinispanRecorder.java
+++ b/extensions/infinispan-client/runtime/src/main/java/io/quarkus/infinispan/client/runtime/InfinispanRecorder.java
@@ -37,7 +37,15 @@ public class InfinispanRecorder {
         return new InfinispanClientSupplier<>(new Function<>() {
             @Override
             public RemoteCacheManager apply(InfinispanClientProducer infinispanClientProducer) {
-                return infinispanClientProducer.getNamedRemoteCacheManager(clientName);
+                RemoteCacheManager result = infinispanClientProducer.getNamedRemoteCacheManager(clientName);
+                if (result == null) {
+                    if (clientName.startsWith(InfinispanClientUtil.DEFAULT_INFINISPAN_CLIENT_NAME)) {
+                        throw new IllegalStateException("No configuration found for the default Infinispan client");
+                    } else {
+                        throw new IllegalStateException("No configuration found for Infinispan client '" + clientName + "'");
+                    }
+                }
+                return result;
             }
         });
     }

